data_management/services/dlu.py

```diff
@@ -0,0 +1,185 @@
+        "%Y-%m-%d %H:%M:%S"
+    )
+    return (
+        dlu_inventory["dluPackageId"],
+        dt_string,
+        dlu_inventory["dluSubmitter"],
+        dlu_inventory["dluTis"],
+        dlu_inventory["dluPackageType"],
+        dlu_inventory["dluSubjectId"],
+        dlu_inventory["dluError"],
+        dlu_inventory["dluLfu"],
+        dlu_inventory["knownSpecimen"],
+        dlu_inventory["redcapId"],
+        dlu_inventory["userPackageReady"],
+        dlu_inventory["dvcValidationComplete"],
+        dlu_inventory["packageValidated"],
+        dlu_inventory["readyToPromoteDlu"],
+        dlu_inventory["promotionDluSucceeded"],
+        dlu_inventory["removedFromGlobus"],
+        dlu_inventory["promotionStatus"],
+        dlu_inventory["notes"],
+    )
+
+
+def dlu_file_dict_to_tuple(dlu_file: dict):
+    return (
+        dlu_file["dluFileName"],
+        dlu_file["dluPackageId"],
+        dlu_file["dluFileId"],
+        dlu_file["dluFileSize"],
+        dlu_file["dluMd5Checksum"],
+    )
+
+
+def create_dest_directory(dest_path: str):
+    if not os.path.exists(dest_path):
+        logger.info("Destination directory " + dest_path + " does not exist. Creating.")
+        os.mkdir(dest_path)
+    else:
+        logger.info("Destination directory exists. Deleting metadata.json if it exists.")
+        if os.path.exists(os.path.join(dest_path, "metadata.json")):
+            os.remove(os.path.join(dest_path, "metadata.json"))
+
+
+def calculate_checksum(file_path: str):
+    with open(file_path, "rb") as f:
+        file_hash = hashlib.md5()
+        while chunk := f.read(8192):
+            file_hash.update(chunk)
+    return file_hash.hexdigest()
+
+
+class DirectoryInfo:
+    def __init__(self, directory_path: str):
+        self.dir_contents = os.listdir(directory_path)
+        self.dir_count = 0
+        self.file_count = 0
+        self.file_details = []
+        self.valid_for_dlu = False
+        self.directory_path = directory_path
+        self.get_directory_information()
+        self.check_if_valid_for_dlu()
+
+    def get_directory_information(self):
+        for item in self.dir_contents:
+            full_path = os.path.join(self.directory_path, item)
+            if os.path.isdir(full_path):
+                self.dir_count += 1
+            else:
+                self.file_count += 1
+                self.file_details.append({
+                    "name": item,
+                    "path": full_path,
+                    "checksum": calculate_checksum(full_path),
+                    "size": os.path.getsize(full_path)
+                })
+
+    def check_if_valid_for_dlu(self):
+        directory_not_empty = len(self.dir_contents) != 0
+        # Nothing but a subdir
+        if self.dir_count == 1 and self.file_count == 1:
+            self.valid_for_dlu = True
+        # Items and no subdirectories
+        elif self.dir_count == 0 and directory_not_empty:

```

Comment:
I am a little confused as to when this check happens. Does this happen before we move data from Globus? If so, shouldn't it be False if there are files in the directory? Or is this a check that happens after a move and we are making sure we see files in the directory?
If it happens after the move, shouldn't we check that it has all of the files that were in Globus?

Reply:
Oh, nvm...I see. This is used to check the Globus directory to make sure it has files and not too many subdirectories. All is well.